lib/models/branch.js

```diff
@@ -67,6 +67,23 @@ export default class Branch {
     return this.getRemoteRef().replace(/^(refs\/)?((heads|remotes)\/)?/, '');
   }
 
+  getRefSpec(action) {
+    if (this.isRemoteTracking()) {
+      return '';
+    }
+    const remoteBranch = action === 'PUSH' ? this.push : this.upstream;
+    const remoteBranchName = remoteBranch.getShortRemoteRef();
+    const localBranchName = this.getName();
+    if (remoteBranchName && remoteBranchName !== localBranchName) {
+      if (action === 'PUSH') {
+        return `${localBranchName}:${remoteBranchName}`;
+      } else if (action === 'PULL') {
+        return `${remoteBranchName}:${localBranchName}`;
+      }
```

Comment:
Using strings for enumerated values always makes me a bit nervous - it feels like it's easy to typo them somewhere (or miss a callsite when refactoring, say) and get surprising behavior. Sometimes I'll use Symbols instead, or objects with a `.when()` method, to push the "getting a valid action" work down into the language more.

With that said: the real answer here is "port our package to TypeScript" anyway. So this is just fine for now :smile: